class.phpmailer.php

```diff
@@ -840,6 +840,13 @@ protected function addAnAddress($kind, $address, $name = '')
             }
             return false;
         }
+        if ($this->hasMultiBytes($address) && function_exists('idn_to_ascii')) {
+            // can't use $this->parseAddresses, because it calls validateAddress, which will fail here
+            list($U,$D) = explode('@',$address);
```

Comment:
actually, parseAddresses and imap_rfc822_parse_adrlist handle email addresses with name etc. Here we only deal with an email address, so the explode is fine.